src/azure-cli-core/azure/cli/core/commands/parameters.py

```diff
@@ -103,6 +103,12 @@ def model_choice_list(resource_type, model_name):
     return enum_choice_list(model) if model else {}
 
 
+def create_enum_type(choices):
+    """ Creates the type for a supplied enum type or list of strings. """
+    def _type(value):
+        return next((x for x in choices if x.lower() == value.lower()), value) if value else value
```

Comment:
I made this because `enum_choice_list` was causing the choices to be sorted in alphabetical order in help text, which for this particular enum I didn't want. So this function allows for arg validation while allowing me to write the help text myself.